### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by tmigot**
The tutorial [to be done] will explain the kwargs, callback ...

### CODE REVIEW

**Comment by dpo** [comment]
Any idea what's going on with the nightly failure?

File: README.md [hunk 1]
```diff
@@ -10,8 +10,54 @@
 [![Cirrus CI - Base Branch Build Status](https://img.shields.io/cirrus/github/JuliaSmoothOptimizers/JSOSolvers.jl?logo=Cirrus%20CI)](https://cirrus-ci.com/github/JuliaSmoothOptimizers/JSOSolvers.jl)
 
 This package provides optimization solvers curated by the JuliaSmoothOptimizers
-organization.
+organization for unconstrained optimization
+
+    min f(x)
+
+or bound-constrained optimization
```
Comment (dpo) [inline README.md:17]:
```suggestion
and bound-constrained optimization
```

File: README.md [hunk 2]
```diff
@@ -10,8 +10,54 @@
...
+
+    min f(x)     s.t.  ℓ ≤ x ≤ u
+
+This package contains the implementation of four algorithms that are classical for unconstrained/bound-constrained nonlinear optimization:
```
Comment (dpo) [inline README.md:21]:
```suggestion
This package provides an implementation of four classic algorithms for unconstrained/bound-constrained nonlinear optimization:
```

File: README.md [hunk 3]
```diff
@@ -10,8 +10,54 @@
...
+- `lbfgs`: An implementation of a limited memory BFGS line-search method for unconstrained minimization.
```
Comment (dpo) [inline README.md:22]:
```suggestion
- `lbfgs`: an implementation of a limited-memory BFGS line-search method for unconstrained minimization;
```

File: README.md [hunk 4]
```diff
@@ -10,8 +10,54 @@
...
+- `R2`: A first-order quadratic regularization method for unconstrained optimization.
```
Comment (dpo) [inline README.md:23]:
```suggestion
- `R2`: a first-order quadratic regularization method for unconstrained optimization;
```

File: README.md [hunk 5]
```diff
@@ -10,8 +10,54 @@
...
+- `tron`: A pure Julia implementation of TRON a trust-region solver for bound-constrained optimization described in
```
Comment (dpo) [inline README.md:24]:
```suggestion
- `tron`: a pure Julia implementation of TRON, a trust-region solver for bound-constrained optimization described in
```

File: README.md [hunk 6]
```diff
@@ -10,8 +10,54 @@
...
+
+    >  Chih-Jen Lin and Jorge J. Moré, *Newton's Method for Large Bound-Constrained
+    >  Optimization Problems*, SIAM J. Optim., 9(4), 1100–1127, 1999.
+    >  DOI: [10.1137/S1052623498345075](https://www.doi.org/10.1137/S1052623498345075)
+
+    The package also contains a variant for nonlinear least-squares.
```
Comment (dpo) [inline README.md:30]:
```suggestion
    as well as a variant for nonlinear least-squares;
```

File: README.md [hunk 7]
```diff
@@ -10,8 +10,54 @@
...
+- `trunk`: A trust-region solver for unconstrained optimization using exact second derivatives. This implementation follows the description given in
```
Comment (dpo) [inline README.md:31]:
```suggestion
- `trunk`: a trust-region solver for unconstrained optimization using exact second derivatives. Our implementation follows the description given in
```

**Review by dpo — CHANGES_REQUESTED** [review]

**Comment by tmigot** [comment]
> Any idea what's going on with the nightly failure?

This happens randomly but regularly with this package with all the versions. It looks like some numerical instability that we never really investigated

File: README.md [hunk 8]
```diff
@@ -10,8 +10,54 @@
...
-If you use JSOSolvers.jl in your work, please cite using the format given in [CITATION.bib](CITATION.bib).
\ No newline at end of file
+If you use JSOSolvers.jl in your work, please cite using the format given in [CITATION.bib](CITATION.bib).
+
+# Bug reports and discussions
+
+If you think you found a bug, feel free to open an [issue](https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl/issues).
+Focused suggestions and requests can also be opened as issues. Before opening a pull request, start an issue or a discussion on the topic, please.
+
+If you want to ask a question not suited for a bug report, feel free to start a discussion [here](https://github.com/JuliaSmoothOptimizers/Organization/discussions). This forum is for general discussion about this repository and the [JuliaSmoothOptimizers](https://github.com/JuliaSmoothOptimizers), so questions about any of our packages are welcome.
```
Comment (dpo) [inline README.md:63]: every README should mention this!

**Review by dpo — APPROVED** [review]

**Pull request merged**